crates/matrix-sdk/src/room/mod.rs

```diff
@@ -2976,6 +2979,60 @@ impl Room {
             .await?;
         Ok(())
     }
+
+    /// Subscribe to live location sharing events for this room.
+    ///
+    /// The returned receiver will receive a new event for each sync response
+    /// that contains a 'm.beacon' event.
+    pub fn subscribe_to_live_location_shares(
+        &self,
+    ) -> (JoinHandle<()>, broadcast::Receiver<LiveLocationShare>) {
+        let (sender, receiver) = broadcast::channel(16);
+
+        let client = self.client.clone();
+        let room_id = self.room_id().to_owned();
+
+        let handle: JoinHandle<()> = tokio::spawn(async move {
+            let beacon_event_handler_handle = client.add_room_event_handler(&room_id, {
+                move |event: OriginalSyncBeaconEvent| async move {
+                    let live_location_share = LiveLocationShare {
+                        user_id: event.sender,
+                        last_location: LastLocation {
+                            location: event.content.location,
+                            ts: event.content.ts,
+                        },
```

Comment:
I'm having trouble adding `beacon_info` to the struct with `self.get_user_beacon_info()` due to borrowed data issues, and I'm still learning the nuances of Rust tasks.

Reply:
Do you still have troubles?

Reply:
Yes, I'm having trouble integrating the [`beacon_info`](https://github.com/ruma/ruma/pull/1722/files#diff-7c2cc8144323bee966e5e97a07ad9b7d556ec8697570d12b5a1a01c9da10452eR75) into my struct, which contains information about the sender, timeout, and `is_live()`. I've commented out `beacon_info` in the struct until I can better understand how to include it.

Each user sharing their live location in a room is [stored here](https://github.com/torrybr/matrix-rust-sdk/blob/5ba90611b43915926a37be83f3de1d4d53bca6bd/crates/matrix-sdk-base/src/rooms/mod.rs#L88). My idea is to, on every beacon update, retrieve the related `beacon_info` and send it to the receiver. This is why I updated `get_user_beacon_info` to obtain `beacon_info` based on the sender.

Reply:
Is it the caller's responsibility to retrieve the `beacon_info` based on the user ID provided in the `LiveLocationShare` struct, or should this function handle the retrieval of `beacon_info` as part of the event processing?